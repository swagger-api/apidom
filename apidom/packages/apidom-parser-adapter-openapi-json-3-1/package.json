{
  "name": "apidom-parser-adapter-openapi-json-3-1",
  "version": "0.1.0",
  "description": "OpenAPI 3.1.x JSON adapter for ApiDOM parser",
  "main": "cjs/adapter.js",
  "module": "es/adapter.js",
  "jsnext:main": "es/adapter.js",
  "unpkg": "dist/apidom-parser-apdater-openapi-json-3-1.browser.min.js",
  "types": "types/adapter.d.ts",
  "scripts": {
    "build": "npm run clean && npm run typescript:declaration && npm run build:es && npm run build:cjs && npm run build:umd:browser",
    "build:es": "cross-env BABEL_ENV=es babel src --out-dir es --extensions '.ts' --root-mode 'upward'",
    "build:cjs": "cross-env BABEL_ENV=cjs babel src --out-dir cjs --extensions '.ts' --root-mode 'upward'",
    "build:umd:browser": "cross-env BABEL_ENV=browser webpack --config config/webpack/browser.config.js --progress",
    "lint": "eslint ./",
    "lint:fix": "eslint ./ --fix",
    "clean": "rimraf ./es ./cjs ./dist ./types",
    "typescript:check-types": "tsc --noEmit",
    "typescript:declaration": "tsc -p declaration.tsconfig.json",
    "test": "cross-env BABEL_ENV=cjs mocha",
    "perf": "cross-env BABEL_ENV=cjs node ./test/perf/index.js",
    "perf:lexical-analysis": "cross-env BABEL_ENV=cjs node ./test/perf/lexical-analysis.js",
    "perf:syntactic-analysis": "cross-env BABEL_ENV=cjs node ./test/perf/syntactic-analysis.js",
    "perf:refract": "cross-env BABEL_ENV=cjs node ./test/perf/refract.js",
    "perf:parse": "cross-env BABEL_ENV=cjs node ./test/perf/parse.js"
  },
  "author": "Vladimir Gorej",
  "license": "Apache-2.0",
  "dependencies": {
    "@babel/runtime-corejs3": "=7.15.4",
    "@types/ramda": "=0.27.44",
    "apidom": "*",
    "apidom-ns-openapi-3-1": "*",
    "apidom-parser-adapter-json": "*",
    "ramda": "=0.27.1",
    "ramda-adjunct": "=2.33.0"
  }
}
