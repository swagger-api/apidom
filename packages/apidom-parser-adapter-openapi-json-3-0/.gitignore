/dist
/es
/cjs
/types
/NOTICE
/swagger-api-apidom-parser-adapter-openapi-json-3-0-*.tgz
