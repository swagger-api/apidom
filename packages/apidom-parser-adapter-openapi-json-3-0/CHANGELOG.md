# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.55.0](https://github.com/swagger-api/apidom/compare/v0.54.0...v0.55.0) (2022-11-04)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-0

# [0.54.0](https://github.com/swagger-api/apidom/compare/v0.53.0...v0.54.0) (2022-10-28)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-0

# [0.53.0](https://github.com/swagger-api/apidom/compare/v0.52.0...v0.53.0) (2022-10-27)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-0

# [0.52.0](https://github.com/swagger-api/apidom/compare/v0.51.1...v0.52.0) (2022-10-25)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-0

## [0.51.1](https://github.com/swagger-api/apidom/compare/v0.51.0...v0.51.1) (2022-10-21)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-0

# [0.51.0](https://github.com/swagger-api/apidom/compare/v0.50.0...v0.51.0) (2022-10-20)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-0

# [0.50.0](https://github.com/swagger-api/apidom/compare/v0.49.0...v0.50.0) (2022-10-13)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-0

# [0.49.0](https://github.com/swagger-api/apidom/compare/v0.48.0...v0.49.0) (2022-10-11)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-0

# [0.48.0](https://github.com/swagger-api/apidom/compare/v0.47.0...v0.48.0) (2022-10-04)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-0

# [0.47.0](https://github.com/swagger-api/apidom/compare/v0.46.0...v0.47.0) (2022-09-29)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-0

# [0.46.0](https://github.com/swagger-api/apidom/compare/v0.45.0...v0.46.0) (2022-09-21)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-0

# [0.45.0](https://github.com/swagger-api/apidom/compare/v0.44.0...v0.45.0) (2022-09-13)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-0

# [0.44.0](https://github.com/swagger-api/apidom/compare/v0.43.0...v0.44.0) (2022-09-05)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-0

# [0.43.0](https://github.com/swagger-api/apidom/compare/v0.42.0...v0.43.0) (2022-08-31)

### Features

- add JSON parser adapter for OpenAPI 3.0.x namespacd ([#1948](https://github.com/swagger-api/apidom/issues/1948)) ([ce254b2](https://github.com/swagger-api/apidom/commit/ce254b236082667e982b99d6fa32f565ffae1730)), closes [#1892](https://github.com/swagger-api/apidom/issues/1892)
- add YAML parser adapter for OpenAPI 3.0.x namespace ([#1950](https://github.com/swagger-api/apidom/issues/1950)) ([ae469b4](https://github.com/swagger-api/apidom/commit/ae469b48891d529a62d51fdfea937d88bdf416e1)), closes [#1893](https://github.com/swagger-api/apidom/issues/1893)
