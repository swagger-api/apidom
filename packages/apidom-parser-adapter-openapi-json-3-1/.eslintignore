/dist
/es
/cjs
/types
/config
/.nyc_output
/node_modules
/swagger-api-apidom-adapter-openapi-json-3-1-*.tgz
