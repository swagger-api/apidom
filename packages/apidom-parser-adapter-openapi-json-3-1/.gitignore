/dist
/es
/cjs
/types
/NOTICE
/swagger-api-apidom-parser-adapter-openapi-json-3-1-*.tgz
