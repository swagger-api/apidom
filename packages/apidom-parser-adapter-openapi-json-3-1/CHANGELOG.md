# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.30.0](https://github.com/swagger-api/apidom/compare/v0.29.1...v0.30.0) (2022-06-07)


### Bug Fixes

* **media-types:** unify how media types are defined ([#1550](https://github.com/swagger-api/apidom/issues/1550)) ([dcb45ce](https://github.com/swagger-api/apidom/commit/dcb45ceba18e5c73dd9e53bea53b302860a214db)), closes [#1549](https://github.com/swagger-api/apidom/issues/1549)
* **OpenAPI:** detect OpenAPI in different formats properly ([#1552](https://github.com/swagger-api/apidom/issues/1552)) ([b21e81e](https://github.com/swagger-api/apidom/commit/b21e81ead95b7de5cc8b5e4161f6096c79a62c0b)), closes [#422](https://github.com/swagger-api/apidom/issues/422)





## [0.29.1](https://github.com/swagger-api/apidom/compare/v0.29.0...v0.29.1) (2022-06-03)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.29.0](https://github.com/swagger-api/apidom/compare/v0.28.0...v0.29.0) (2022-05-27)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.28.0](https://github.com/swagger-api/apidom/compare/v0.27.0...v0.28.0) (2022-05-22)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.27.0](https://github.com/swagger-api/apidom/compare/v0.26.0...v0.27.0) (2022-05-20)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.26.0](https://github.com/swagger-api/apidom/compare/v0.25.0...v0.26.0) (2022-05-20)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.25.0](https://github.com/swagger-api/apidom/compare/v0.24.1...v0.25.0) (2022-05-13)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.24.0](https://github.com/swagger-api/apidom/compare/v0.23.0...v0.24.0) (2022-05-03)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.23.0](https://github.com/swagger-api/apidom/compare/v0.22.0...v0.23.0) (2022-04-25)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.22.0](https://github.com/swagger-api/apidom/compare/v0.21.0...v0.22.0) (2022-04-21)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.21.0](https://github.com/swagger-api/apidom/compare/v0.20.1...v0.21.0) (2022-04-18)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.20.0](https://github.com/swagger-api/apidom/compare/v0.19.0...v0.20.0) (2022-04-11)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.19.0](https://github.com/swagger-api/apidom/compare/v0.18.1...v0.19.0) (2022-03-10)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





## [0.18.1](https://github.com/swagger-api/apidom/compare/v0.18.0...v0.18.1) (2022-02-16)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.18.0](https://github.com/swagger-api/apidom/compare/v0.17.0...v0.18.0) (2022-02-16)


### Features

* **apidom-ast:** convert to ESM ([#1175](https://github.com/swagger-api/apidom/issues/1175)) ([75231d6](https://github.com/swagger-api/apidom/commit/75231d690e485c9e96da5fb806fc35712018f81c)), closes [#930](https://github.com/swagger-api/apidom/issues/930)
* convert 3 additional packages to ESM modules ([#1185](https://github.com/swagger-api/apidom/issues/1185)) ([b1f314b](https://github.com/swagger-api/apidom/commit/b1f314bb68cb57e18d9e925e0c5191c31faba49a)), closes [#930](https://github.com/swagger-api/apidom/issues/930)
* **parser-adapter-opneapi-json-3-1:** convert to ESM ([f6de4df](https://github.com/swagger-api/apidom/commit/f6de4df8c23bb6fd403a41ef36988c9ea30f7aa9)), closes [#930](https://github.com/swagger-api/apidom/issues/930)





# [0.17.0](https://github.com/swagger-api/apidom/compare/v0.16.0...v0.17.0) (2022-01-14)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.16.0](https://github.com/swagger-api/apidom/compare/v0.15.1...v0.16.0) (2022-01-12)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





## [0.15.1](https://github.com/swagger-api/apidom/compare/v0.15.0...v0.15.1) (2021-12-29)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.15.0](https://github.com/swagger-api/apidom/compare/v0.14.0...v0.15.0) (2021-12-10)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.14.0](https://github.com/swagger-api/apidom/compare/v0.13.0...v0.14.0) (2021-12-08)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.13.0](https://github.com/swagger-api/apidom/compare/v0.12.0...v0.13.0) (2021-12-06)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.12.0](https://github.com/swagger-api/apidom/compare/v0.11.0...v0.12.0) (2021-12-04)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.11.0](https://github.com/swagger-api/apidom/compare/v0.10.0...v0.11.0) (2021-12-02)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.10.0](https://github.com/swagger-api/apidom/compare/v0.9.0...v0.10.0) (2021-11-25)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.9.0](https://github.com/swagger-api/apidom/compare/v0.8.0...v0.9.0) (2021-11-17)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.8.0](https://github.com/swagger-api/apidom/compare/v0.7.0...v0.8.0) (2021-11-15)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.7.0](https://github.com/swagger-api/apidom/compare/v0.6.0...v0.7.0) (2021-11-08)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.6.0](https://github.com/swagger-api/apidom/compare/v0.5.1...v0.6.0) (2021-11-04)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





## [0.5.1](https://github.com/swagger-api/apidom/compare/v0.5.0...v0.5.1) (2021-11-01)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.5.0](https://github.com/swagger-api/apidom/compare/v0.4.0...v0.5.0) (2021-10-26)


### Performance Improvements

* optimize JSON direct syntactic analysis ([5b5887e](https://github.com/swagger-api/apidom/commit/5b5887ec1d38a0cb3b4c542997818f39cda84108)), closes [#691](https://github.com/swagger-api/apidom/issues/691)





# [0.4.0](https://github.com/swagger-api/apidom/compare/v0.3.0...v0.4.0) (2021-10-22)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# [0.3.0](https://github.com/swagger-api/apidom/compare/v0.2.1...v0.3.0) (2021-10-05)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





## [0.2.1](https://github.com/swagger-api/apidom/compare/v0.2.0...v0.2.1) (2021-09-30)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# 0.2.0 (2021-09-28)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1





# 0.1.0 (2021-09-28)

**Note:** Version bump only for package @swagger-api/apidom-parser-adapter-openapi-json-3-1
