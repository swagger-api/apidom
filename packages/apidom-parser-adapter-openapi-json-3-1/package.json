{
  "name": "@swagger-api/apidom-parser-adapter-openapi-json-3-1",
  "version": "0.66.0",
  "description": "Parser adapter for parsing JSON documents into OpenAPI 3.1.x namespace.",
  "publishConfig": {
    "registry": "https://npm.pkg.github.com"
  },
  "type": "module",
  "unpkg": "./dist/apidom-parser-apdater-openapi-json-3-1.browser.min.js",
  "main": "./cjs/adapter.cjs",
  "exports": {
    "types": "./types/dist.d.ts",
    "import": "./es/adapter.js",
    "require": "./cjs/adapter.cjs"
  },
  "types": "./types/dist.d.ts",
  "scripts": {
    "build": "npm run clean && run-p --max-parallel ${CPU_CORES:-2} typescript:declaration build:es build:cjs build:umd:browser",
    "build:es": "cross-env BABEL_ENV=es babel src --out-dir es --extensions '.ts' --root-mode 'upward'",
    "build:cjs": "cross-env BABEL_ENV=cjs babel src --out-dir cjs --extensions '.ts' --out-file-extension '.cjs' --root-mode 'upward'",
    "build:umd:browser": "cross-env BABEL_ENV=browser BROWSERSLIST_ENV=production webpack --config config/webpack/browser.config.js --progress",
    "lint": "eslint ./",
    "lint:fix": "eslint ./ --fix",
    "clean": "rimraf ./es ./cjs ./dist ./types",
    "typescript:check-types": "tsc --noEmit",
    "typescript:declaration": "tsc -p declaration.tsconfig.json && rollup -c config/rollup/types.dist.js",
    "test": "cross-env NODE_ENV=test BABEL_ENV=cjs mocha",
    "perf": "cross-env NODE_ENV=test BABEL_ENV=cjs node ./test/perf/index.cjs",
    "perf:lexical-analysis": "cross-env NODE_ENV=test BABEL_ENV=cjs node ./test/perf/lexical-analysis.cjs",
    "perf:syntactic-analysis": "cross-env NODE_ENV=test BABEL_ENV=cjs node ./test/perf/syntactic-analysis.cjs",
    "perf:refract": "cross-env NODE_ENV=test BABEL_ENV=cjs node ./test/perf/refract.cjs",
    "perf:parse": "cross-env NODE_ENV=test BABEL_ENV=cjs node ./test/perf/parse.cjs",
    "prepack": "copyfiles -u 3 ../../LICENSES/* LICENSES && copyfiles -u 2 ../../NOTICE .",
    "postpack": "rimraf NOTICE LICENSES"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/swagger-api/apidom.git"
  },
  "author": "Vladimir Gorej",
  "license": "Apache-2.0",
  "dependencies": {
    "@babel/runtime-corejs3": "=7.20.7",
    "@swagger-api/apidom-core": "^0.66.0",
    "@swagger-api/apidom-ns-openapi-3-1": "^0.66.0",
    "@swagger-api/apidom-parser-adapter-json": "^0.66.0",
    "@types/ramda": "=0.28.22",
    "ramda": "=0.28.0",
    "ramda-adjunct": "=3.4.0"
  },
  "files": [
    "cjs/",
    "dist/",
    "es/",
    "types/dist.d.ts",
    "LICENSES",
    "NOTICE",
    "README.md",
    "CHANGELOG.md"
  ]
}
