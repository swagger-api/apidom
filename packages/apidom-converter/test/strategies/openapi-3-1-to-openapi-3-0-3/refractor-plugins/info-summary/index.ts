import path from 'node:path';
import { expect, assert } from 'chai';
import { mediaTypes as openAPI31MediaTypes } from '@swagger-api/apidom-parser-adapter-openapi-json-3-1';
import { mediaTypes as openAPI30MediaTypes } from '@swagger-api/apidom-parser-adapter-openapi-json-3-0';
import {
  AnnotationElement,
  ParseResultElement,
  toJSON,
  includesClasses,
} from '@swagger-api/apidom-core';

import convert from '../../../../../src';

describe('converter', function () {
  context('strategies', function () {
    context('openapi-3-1-to-openapi-3-0-3', function () {
      context('info-summary', function () {
        const fixturePath = path.join(__dirname, 'fixtures', 'info-summary.json');
        let convertedParseResult: ParseResultElement;

        beforeEach(async function () {
          convertedParseResult = await convert(fixturePath, {
            convert: {
              sourceMediaType: openAPI31MediaTypes.findBy('3.1.0', 'json'),
              targetMediaType: openAPI30MediaTypes.findBy('3.0.3', 'json'),
            },
          });
        });

        specify('should remove Info.summary field', async function () {
          expect(toJSON(convertedParseResult.api!, undefined, 2)).toMatchSnapshot();
        });

        specify('should create WARNING annotation', async function () {
          const annotations = Array.from(convertedParseResult.annotations);
          const annotation = annotations.find((a: AnnotationElement) =>
            a.code?.equals('info-summary'),
          );

          assert.isDefined(annotation);
          assert.isTrue(includesClasses(['warning'], annotation));
        });
      });
    });
  });
});
