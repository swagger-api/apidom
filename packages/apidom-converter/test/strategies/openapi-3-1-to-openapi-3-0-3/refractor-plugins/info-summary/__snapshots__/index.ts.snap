// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converter strategies openapi-3-1-to-openapi-3-0-3 info-summary should remove Info.summary field 1`] = `
{
  "openapi": "3.0.3",
  "info": {
    "version": "1.0.0",
    "title": "A title"
  }
}
`;
